### PR TITLE
feat: endpoint to create blog posts

### PR DESCRIPTION
**Pull request by chionz**
## Description

Added a new API endpoint to allow only a super admin to create a blog post. This includes implementing token-based authentication to ensure only users with the super admin role can access this functionality. The endpoint verifies the presence of a valid token in the Authorization header and validates the request body to ensure the required fields are provided and correctly formatted.

## Related Issue (Link to issue ticket)

[[FEAT]: Blog - Creating a Blog Post #113](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/113)

## Motivation and Context

This change is required to restrict blog post creation to super admins only, enhancing security and ensuring only authorized personnel can create blog content. It solves the problem of unauthorized access to the blog creation endpoint.

## How Has This Been Tested?

Unit tests were implemented to cover various scenarios, including:
- Valid authorization token in the header
- Invalid authorization token
- Authorization token expiry time
- Invalid request bodies (missing required fields)
- Valid request bodies

Tests were run in a local development environment using a combination of automated testing tools and manual testing via Postman to ensure all functionalities work as expected and to verify the correct status codes and responses are returned for different cases.

## Screenshots (if appropriate - Postman, etc):

Screenshot from 2024-07-24 02-08-37

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: api/v1/routes/blog.py [hunk 1]
```diff
@@ -3,15 +3,29 @@
...
 from api.db.database import get_db
 from api.v1.models.user import User
-from api.utils.dependencies import get_current_user
+from api.utils.dependencies import get_current_user, get_super_admin
 from api.v1.services.blog import BlogService
 from fastapi import APIRouter, Depends, HTTPException, status
 from typing import List
 from api.v1.models.blog import Blog
 from api.v1.schemas.blog import BlogResponse
+from api.v1.schemas.blog import BlogCreate
```
Comment (Goketech) [inline api/v1/routes/blog.py:12]: change this

from api.v1.schemas.blog import BlogResponse
from api.v1.schemas.blog import BlogCreate

to 

from api.v1.schemas.blog import BlogResponse, BlogCreate

**Review by Goketech — CHANGES_REQUESTED** [review]

**Comment by chionz** [comment]
@Goketech done that now

**Review by Goketech — APPROVED** [review]

**Comment by chionz** [comment]
@bolexs I have done just that now

**Review by Goketech — APPROVED** [review]

File: api/v1/routes/blog.py [hunk 2]
```diff
@@ -2,17 +2,29 @@
...
+from fastapi import APIRouter, Depends, HTTPException, status
+from typing import List
+from api.v1.models.blog import Blog
+from api.v1.schemas.blog import BlogResponse, BlogCreate, BlogPostResponse, BlogRequest, BlogUpdateResponseModel
 
 blog = APIRouter(prefix="/blogs", tags=["Blog"])
 
+@blog.post("/api/v1/blogs")
+def create_blog(blog: BlogCreate, db: Session = Depends(get_db), current_user: User = Depends(get_super_admin)):
+    if not current_user:
```
Comment (joboy-dev) [inline api/v1/routes/blog.py:18]: There's no need to check for authentication 
It's already handled by the get current user function

**Review by joboy-dev — APPROVED** [review]

File: api/v1/routes/blog.py [hunk 3]
```diff
@@ -2,17 +2,29 @@
...
+        raise HTTPException(status_code=401, detail="You are not Authorized")
+    blog_service = BlogService(db)
+    new_blogpost = blog_service.create(db=db, schema=blog, author_id=current_user.id)
```
Comment (zxenonx) [inline api/v1/routes/blog.py:21]: blog_post is OK.

**Review by zxenonx — APPROVED** [review]
Thanks @chionz

**Pull request merged**